test/python/test_rticonnextdds_discovery.py

```diff
@@ -177,19 +186,30 @@ def test_unmatched_output(self, discovery_reader_only_input):
     assert the_output is not None
 
     # Check that matching occurs
-    change_in_matches = discovery_reader_only_input.wait_for_match()
+    change_in_matches = discovery_reader_only_input.wait_for_publications()
     assert change_in_matches == 1
-    matches = discovery_reader_only_input.get_matched_outputs()
+    matches = discovery_reader_only_input.get_matched_publications()
     assert {'name': 'TestWriter'} in matches
-    change_in_matches = the_output.wait_for_match()
+    change_in_matches = the_output.wait_for_subscriptions()
     assert change_in_matches == 1
-    matches = the_output.get_matched_inputs()
+    matches = the_output.get_matched_subscriptions()
     assert {'name': 'TestReader'} in matches
 
     # If we now delete the reader_connector object which we created above, they
     # will unmatch
     writer_connector.close()
-    change_in_matches = discovery_reader_only_input.wait_for_match()
+    change_in_matches = discovery_reader_only_input.wait_for_publications()
     assert change_in_matches == -1
-    matches = discovery_reader_only_input.get_matched_outputs()
-    assert matches is None
+    matches = discovery_reader_only_input.get_matched_publications()
+    assert len(matches) == 0
+
+  def test_no_entity_names(self, discovery_connector_no_entity_names):
+    the_output = discovery_connector_no_entity_names.get_output("MyPublisher::MyWriter")
+    # Ensure that the entities match
+    change_in_subs = the_output.wait_for_subscriptions(-1)
+    assert change_in_subs == 1
+
+    # Get the entity names from the matched publication and subscriptions
+    matched_subs = the_output.get_matched_subscriptions()
+    # Since the entities used in this test do not set a entity_name QoS, check that no name is present
+    assert matched_subs == [{'name': ''}]
```

Comment:
we discussed adding a test where we create non-Connector readers and writers, in which case the pub/sub name may not be set.

Reply:
Added.